src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -89,5 +153,32 @@ internal bool IsScriptFileDetermined(FunctionMetadata functionMetadata)
             }
             return true;
         }
+
+        private void AddMetadataFromCustomProviders(List<FunctionMetadata> functionMetadataList)
+        {
+            var functionProviderTasks = new List<Task<ImmutableArray<FunctionMetadata>>>();
+            foreach (var functionProvider in _functionProviders)
+            {
+                functionProviderTasks.Add(functionProvider.GetFunctionMetadataAsync());
+            }
+
+            var functionMetadataListArray = Task.WhenAll(functionProviderTasks).GetAwaiter().GetResult();
+
+            foreach (var someArray in functionMetadataListArray)
+            {
+                if (!someArray.IsDefaultOrEmpty)
```

Comment:
Can you remove this DefaultOrEmpty check, and the ToList/ForEach below and just call functionMetadataList.AddRange?

Reply:
With recent changes, we can't do `AddRange` as we need to look at each metadata individually. So with the type as `Immutable`, we'd have to continue this way.